### PR TITLE
vue support

### PR DESCRIPTION
**Pull request by Niedzwiedzw**

### CODE REVIEW

File: rc/vue.kak [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+try %{
+    require-module css
+    require-module javascript
+}
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/html regions
```
Comment (krobelus) [inline rc/vue.kak:45]: These highlighters and the commands below will clash with the ones from html.kak.
This happens if both a Vue file and an HTML file are edited in the same session.

This one should probably be renamed to `shared/vue`

File: rc/vue.kak [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+hook global BufCreate .*\.xml %{
+    set-option buffer filetype xml
+}
+add-highlighter shared/html/comment region <!--     -->                  fill comment
```
Comment (krobelus) [inline rc/vue.kak:50]: If you want to depend on the html module, you can use
`add-highlighter shared/vue/comment ref html/comment`
and so on. Hard dependencies between modules are not conventional, but for this one it's probably okay.

File: rc/vue.kak (hunk 1 above)
Comment (Niedzwiedzw) [reply rc/vue.kak:45]: @krobelus thanks for your feedback, tbh it's not even working at this point, I have zero knowledge on the syntax that's being used here. For now I'd just like to make it highlight the file as if it was a standard html document, but have it have a type of an HTML file. How would I do that with minimal effort + without copying + pasting the whole html file :D

Comment (krobelus) [reply rc/vue.kak:45]: The easiest way is probably to not change anything, and just use the `html` filetype for `*.vue` files.

I guess your main motivation for this is to be able to use a Vue language server
(https://github.com/kak-lsp/kak-lsp/issues/442).
If that's all, then why not use something like

```toml
[language.html]
filetypes = ["html"]
roots = [".git"]
command = "vue-language-server" # was html-language-server
```

in your `~/.config/kak-lsp/kak-lsp.toml`? Assuming you don't need the html-language-server.

Alternatively, copy/pasting and replacing `html` with `vue`  isn't wrong either. It's not beautiful but it works.
The `c-family.kak` script in Kakoune supports multiple related filetypes, that might be relevant.

Comment (Niedzwiedzw) [reply rc/vue.kak:45]: ... man you are genious

**Comment by lenormf** [comment]
Thanks for the contribution.

If you want to give this another try, you could take inspiration from the [riotjs support script](https://github.com/lenormf/kakoune-extra/blob/master/filetype/riotjs.kak) I wrote a while ago, I think it should be easy to adapt and fairly straight forward.

**Pull request closed without merging**